CMakeLists.txt

```diff
@@ -168,23 +170,35 @@ if(CPU_ONLY)
   target_compile_definitions(MaCh3CompileDefinitions INTERFACE CPU_ONLY)
 endif()
 
-if(MaCh3_GPU_ENABLED)
-  target_compile_definitions(MaCh3CompileDefinitions INTERFACE CUDA)
-  target_compile_definitions(MaCh3CompileDefinitions INTERFACE GPU_ON)
+if(MaCh3_GPU_ENABLED) 
+  target_compile_definitions(MaCh3GPUCompilerOptions INTERFACE CUDA)
+  target_compile_definitions(MaCh3GPUCompilerOptions INTERFACE GPU_ON)
 endif()
 
 if(NOT DEFINED MaCh3_LOW_MEMORY_STRUCTS_ENABLED)
   set(MaCh3_LOW_MEMORY_STRUCTS_ENABLED FALSE)
 endif()
 
 if(MaCh3_LOW_MEMORY_STRUCTS_ENABLED)
-  target_compile_definitions(MaCh3CompileDefinitions INTERFACE _LOW_MEMORY_STRUCTS_)
-endif()
+  target_compile_definitions(MaCh3CompilerOptions INTERFACE _LOW_MEMORY_STRUCTS_)
+  if(MaCh3_GPU_ENABLED)
+    # ETA - think the _LOW_MEMORY_STRUCTS_ also needs this definition
+    target_compile_definitions(MaCh3GPUCompilerOptions INTERFACE _LOW_MEMORY_STRUCTS_)
+  endif(MaCh3_GPU_ENABLED)
+endif(MaCh3_LOW_MEMORY_STRUCTS_ENABLED)
 
 set_target_properties(MaCh3CompilerOptions PROPERTIES EXPORT_NAME CompilerOptions)
-install(TARGETS MaCh3CompilerOptions MaCh3CompileDefinitions
+if(MaCh3_GPU_ENABLED)
+  set_target_properties(MaCh3GPUCompilerOptions PROPERTIES EXPORT_NAME GPUCompilerOptions)
+  #If GPU enable then install both the CPU and GPU options
+  install(TARGETS MaCh3CompilerOptions MaCh3GPUCompilerOptions MaCh3CompileDefinitions
+    EXPORT MaCh3-targets
+    LIBRARY DESTINATION lib/)
+else()
+  install(TARGETS MaCh3CompilerOptions MaCh3CompileDefinitions
     EXPORT MaCh3-targets
     LIBRARY DESTINATION lib/)
+endif()
```

Comment:
this is ugly, I think if MaCh3GPUCompilerOptions  is always apparent will make this bit nicer

Reply:
yep this is fixed now